kernel/src/paging.c

```diff
@@ -0,0 +1,170 @@
+#include "paging.h"
+
+#define create_table_entry(addr) \
+    (((uint64_t)addr & f_paging_ptrmask) | g_pagetable_flags_default | f_paging_present)
+
+uint64_t ofy = 20;
+const uint64_t g_pagetable_flags_default = f_paging_present | f_paging_writable;
+
+uint64_t* get_cr3() {
+    uint64_t* ptr;
+    asm("mov %%cr3, %0" : "=r"(ptr));
+
+    return ptr;
+}
+
+void set_cr3(uint64_t* ptr) {
```

Comment:
Is it intended behavior of set_cr3 to set cr3 equal to the actual pointer and not the value at the address of the pointer. Maybe it should take in the value that you want cr3 to be set to instead?